frame/support/src/traits/preimages.rs

```diff
@@ -0,0 +1,250 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Stuff for dealing with 32-byte hashed preimages.
+
+use codec::{Decode, Encode, EncodeLike, MaxEncodedLen};
+use sp_core::{RuntimeDebug, H256};
+use sp_io::hashing::blake2_256;
+use sp_runtime::{traits::ConstU32, DispatchError};
+use sp_std::borrow::Cow;
+
+pub type Hash = H256;
+pub type BoundedInline = crate::BoundedVec<u8, ConstU32<128>>;
+
+#[derive(
+	Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, scale_info::TypeInfo, RuntimeDebug,
+)]
+pub enum Bounded<T> {
+	/// A Blake2 256 hash with no preimage length. We
+	/// do not support creation of this except for transitioning from legacy state.
+	/// In the future we will make this a pure `Dummy` item storing only the final `dummy` field.
+	Legacy { hash: Hash, dummy: sp_std::marker::PhantomData<T> },
```

Comment:
couldnt we make this the last element in the enum, and then actually just delete it when we are done migrating?

Reply:
We need a phantom item in the enum anyway for `T` so best to put it at index 0 (which as the index of the first item will not change if we want to add more items) rather than index 2 (which as the index of the last item would change if we wanted to add more items).